trade-module/src/main/java/io/edpn/backend/trade/application/mappers/FindCommodityDTOMapper.java

```diff
@@ -0,0 +1,64 @@
+package io.edpn.backend.trade.application.mappers;
+
+import io.edpn.backend.trade.application.dto.CoordinateDTO;
+import io.edpn.backend.trade.application.dto.FindCommodityRequest;
+import io.edpn.backend.trade.application.dto.FindCommodityResponse;
+import io.edpn.backend.trade.domain.model.*;
+import io.edpn.backend.trade.domain.model.System;
+import lombok.RequiredArgsConstructor;
+
+import java.util.Optional;
+
+@RequiredArgsConstructor
+public class FindCommodityDTOMapper {
+    
+    public FindCommodityResponse map(FindCommodity findCommodity){
+        return FindCommodityResponse.builder()
+                .pricesUpdate(findCommodity.getPricesUpdatedAt())
+                .commodityName(findCommodity.getCommodity().getName())
+                .station(mapStation(findCommodity.getStation()))
+                .system(mapSystem(findCommodity.getSystem()))
+                .build();
+    }
+    
+    public FindCommodityFilter map(FindCommodityRequest findCommodityRequest){
+        return FindCommodityFilter.builder()
+                .commodityId(findCommodityRequest.getCommodityId())
+                .xCoordinate(findCommodityRequest.getReferenceLocation().getXCoordinate())
+                .yCoordinate(findCommodityRequest.getReferenceLocation().getYCoordinate())
+                .zCoordinate(findCommodityRequest.getReferenceLocation().getZCoordinate())
+                .includePlanetary(findCommodityRequest.getIncludePlanetary())
+                .includeOdyssey(findCommodityRequest.getIncludePlanetary())
+                .includeFleetCarriers(findCommodityRequest.getIncludeFleetCarriers())
+                .landingPadSize(LandingPadSize.valueOf(findCommodityRequest.getLandingPadSize()))
+                .minSupply(findCommodityRequest.getMinSupply())
+                .minDemand(findCommodityRequest.getMinDemand())
+                .build();
+    }
+    
+    private FindCommodityResponse.Station mapStation(Station station){
+        return FindCommodityResponse.Station.builder()
+                .name(station.getName())
+                .arrivalDistance(station.getArrivalDistance())
+                .landingPadSize(station.getMaxLandingPadSize().name())
+                .planetary(station.getPlanetary())
+                .requireOdyssey(station.getRequireOdyssey())
+                .fleetCarrier(station.isFleetCarrier())
+                .build();
+    }
+    
+    //TODO: nullable coords? fix view?
```

Comment:
If we exclude systems with coords null (I think we should because, where are they??)

then this code is fine, but you could add a defensive programming layer just in case

```suggestion
    private FindCommodityResponse.System mapSystem(System system) {
        return FindCommodityResponse.System.builder()
                .name(system.getName())
                .coordinateDTO(
                        CoordinateDTO.builder()
                                .xCoordinate(Optional.ofNullable(system.getXCoordinate()).orElse(0D))
                                .yCoordinate(Optional.ofNullable(system.getYCoordinate()).orElse(0D))
                                .zCoordinate(Optional.ofNullable(system.getZCoordinate()).orElse(0D))
                                .build()
                )
                .build();
    }
```

or even better:
```suggestion
private FindCommodityResponse.System mapSystem(System system) {
        return FindCommodityResponse.System.builder()
                .name(system.getName())
                .coordinateDTO(coordinateFromSystem(system))
                .build();
    }

    private CoordinateDTO coordinateFromSystem(System system) {
        if (Objects.isNull(system.getXCoordinate())) {
            return null;
        } else {
            return CoordinateDTO.builder()
                    .xCoordinate(system.getXCoordinate())
                    .yCoordinate(system.getYCoordinate())
                    .zCoordinate(system.getZCoordinate())
                    .build();
        }
    }
```